contrib/submit-protocol/src/App/MarketplaceForm.tsx

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import React from "react";
+import {
+  Callout, DefaultButton, Dropdown, DropdownMenuItemType,
+  Fabric, IDropdownOption, Stack, TextField,
+  mergeStyleSets,
+} from "office-ui-fabric-react";
+
+enum MarketplaceUriType {
+  None = "None",
+  GitHub = "GitHub",
+}
+
+interface IMarketplaceProps {
+  uri?: string;
+  uriType?: MarketplaceUriType | string;
+  onSelectProtocol: ((text: string) => void);
+  disabled?: boolean;
+}
+
+interface IMarketplaceState {
+  uri: string;
+  uriType: MarketplaceUriType;
+  selectedProtocol?: IDropdownOption;
```

Comment:
Where is this state used?

Reply:
removed, it was used for selected option in dropdown